src/Message/AbstractRequest.php

```diff
@@ -1,217 +1,217 @@
     }
 
     /**
      * Set single-use token
-     * @param  string $value Token Key
      */
-    public function setSingleUseTokenId($value)
+    public function setSingleUseTokenId(string $value): self
     {
         return $this->setParameter('singleUseTokenId', $value);
     }
 
     /**
      * Get Idempotency Key
-     * @return string Idempotency Key
      */
-    public function getIdempotencyKey()
+    public function getIdempotencyKey(): ?string
     {
         return $this->getParameter('idempotencyKey') ?: uniqid();
     }
 
     /**
      * Set Idempotency Key
-     * @param  string $value Idempotency Key
      */
-    public function setIdempotencyKey($value)
+    public function setIdempotencyKey(string $value): self
     {
         return $this->setParameter('idempotencyKey', $value);
     }
 
-    public function getCustomerNumber()
+    public function getCustomerNumber(): ?string
     {
         return $this->getParameter('customerNumber');
     }
 
-    public function setCustomerNumber($value)
+    public function setCustomerNumber(string $value): self
     {
         return $this->setParameter('customerNumber', $value);
     }
 
-    public function getTransactionType()
+    public function getTransactionType(): ?string
     {
         return $this->getParameter('transactionType');
     }
 
-    public function setTransactionType($value)
+    public function setTransactionType(string $value): self
     {
         return $this->setParameter('transactionType', $value);
     }
 
-    public function getAmount()
+    public function getAmount(): string|Money|null
     {
         return $this->getParameter('amount');
     }
 
-    public function setAmount($value)
+    /**
+     * Retaining the original method signature, although we use the method incorrectly by allowing Money param
+     * @see \Omnipay\Common\Message\AbstractRequest::setAmount
+     * @param string|null|Money $value
+     * @return $this
+     */
+    public function setAmount($value): self
     {
         return $this->setParameter('amount', $value);
     }
 
-    public function getPrincipalAmount()
+    public function getPrincipalAmount(): ?string
     {
         return $this->getParameter('principalAmount');
     }
 
-    public function setPrincipalAmount($value)
+    public function setPrincipalAmount(string $value): self
     {
         return $this->setParameter('principalAmount', $value);
     }
 
-    public function getCurrency()
+    public function getCurrency(): ?string
     {
         // PayWay expects lowercase currency values
         return ($this->getParameter('currency'))
-            ? strtolower($this->getParameter('currency'))
+            ? strtolower((string) $this->getParameter('currency'))
             : null;
     }
 
-    public function setCurrency($value)
+    /**
+     * Retaining the original method signature
+     *
+     * @param string $value
+     * @return $this
+     */
+    public function setCurrency($value): self
```

Comment:
Why can't we use the type in the signature for this one?

Reply:
<img width="916" alt="image" src="https://github.com/user-attachments/assets/afa3e295-7e53-4671-bf16-7c20f9e195bf">
The parent class's declaration has no type hint so if we add the type hint it becomes incompatible.